x-pack/plugins/xpack_main/server/lib/populate_ui_capabilities.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { UICapabilities } from 'ui/capabilities';
+import { Feature } from '../../types';
+
+export function populateUICapabilities(
```

Comment:
**note** I opted for a verbose implementation over a smaller one to hopefully makes this easier to understand and maintain